prog/dftb+/lib_md/tempprofile.F90

```diff
@@ -161,33 +155,55 @@ subroutine TempProfile_next(self)
     case (constProf)
       self%curTemp = self%tempValues(self%iInt)
     case (linProf)
-      if (tChanged) then
-        self%incr = (supVal - subVal) / real(sup - sub, dp)
-      end if
+      self%incr = (supVal - subVal) / real(sup - sub, dp)
       self%curTemp = subVal + self%incr * real(self%iStep - sub, dp)
     case (expProf)
-      if (tChanged) then
-        self%tempValues(self%iInt) = supVal
-        self%tempValues(self%iInt-1) = subVal
-        self%incr = log(supVal/subVal) / real(sup - sub, dp)
-      end if
+      self%tempValues(self%iInt) = supVal
+      self%tempValues(self%iInt-1) = subVal
+      self%incr = log(supVal/subVal) / real(sup - sub, dp)
       self%curTemp = subVal * exp(self%incr * real(self%iStep - sub, dp))
     end select
 
-  end subroutine TempProfile_next
+  end subroutine next
 
 
   !> Returns the current temperature.
-  subroutine TempProfile_getTemperature(self, temp)
+  subroutine getTemperature(self, temp)
 
     !> Pointer to the TempProfile object.
-    type(TTempProfile), intent(in) :: self
+    class(TTempProfile), intent(in) :: self
 
     !> Temperature on return.
     real(dp), intent(out) :: temp
 
     temp = self%curTemp
 
-  end subroutine TempProfile_getTemperature
+  end subroutine getTemperature
+
+
+  !> Maps a (supported) profile name onto integer identifier
+  subroutine identifyTempProfile(iProfile, profileName, success)
+
+    !> Internal profile identifying number
+    integer, intent(out) :: iProfile
+
+    !> Possible profile name
+    character(*), intent(in) :: profileName
+
+    !> was the profile correctly identified
+    logical, intent(out) :: success
+
+    integer :: ii
+
+    success = .false.
+    do ii = 1, size(temperatureProfileNames)
+      if (trim(profileName) == tolower(trim(temperatureProfileNames(ii)))) then
```

Comment:
A `select case` would work as well and avoid the obscure loop here, also we would not have to rely on the fact that the enumerator corresponds to a loop counter.